scripts/base/protocols/dns/main.zeek

```diff
@@ -343,11 +347,20 @@ hook set_session(c: connection, msg: dns_msg, is_query: bool) &priority=5
 		if ( msg$rcode != 0 && msg$num_queries == 0 )
 			c$dns$rejected = T;
 		}
+
+	c$dns$opcode = msg$opcode;
+	if ( msg$is_netbios )
+		if ( msg$opcode >= 5 )
+			c$dns$opcode_name = opcodes[msg$opcode + 0xFFFF];
```

Comment:
I wonder if a separate `netbios_opcodes` table would be easier.

This is pretty tricky to grasp and I'm actually confused if it's correct. Can the `0xFFFF5` from the table ever happen? On first sight, `5 + 0xFFFF`, but that's `'0x100fe'` and not `0xFFFF5`.

If there's separate tables, then `is_netbios` would suffice to distinguish which one to use and adding entries via `redef` wouldn't need the `0xFFFF` tricks.